tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -1330,13 +1330,27 @@ uint32_t SPIRVEmitter::doConditionalOperator(const ConditionalOperator *expr) {
   return theBuilder.createSelect(type, condition, trueBranch, falseBranch);
 }
 
-uint32_t SPIRVEmitter::processBufferLoad(const Expr *object,
-                                         const Expr *location) {
+uint32_t SPIRVEmitter::processBufferTextureLoad(const Expr *object,
+                                                const Expr *location,
+                                                uint32_t constOffset,
+                                                uint32_t varOffset) {
   // Loading for Buffer and RWBuffer translates to an OpImageFetch.
   // The result type of an OpImageFetch must be a vec4 of float or int.
   const auto type = object->getType();
-  const uint32_t objectId = doExpr(object);
+  assert(TypeTranslator::isBuffer(type) || TypeTranslator::isRWBuffer(type) ||
+         TypeTranslator::isTexture(type));
+  const bool doFetch =
+      TypeTranslator::isBuffer(type) || TypeTranslator::isTexture(type);
+  const uint32_t objectId = loadIfGLValue(object);
   const uint32_t locationId = doExpr(location);
+
+  // For Buffers/RWBuffers, the location is just an int, which should be used as
+  // the coordinate argument. Textures require an additional processing.
```

Comment:
I'd suggest to have the original comment previously at line 1641 for textures.

Reply:
Oh yes, Done